coderd/aitasks.go

```diff
@@ -302,6 +302,35 @@ func taskFromDBTaskAndWorkspace(dbTask database.Task, ws codersdk.Workspace) cod
 		}
 	}
 
+	// If no valid agent state was found for the current build and the task is initializing,
+	// provide a descriptive initialization message.
+	if currentState == nil && codersdk.TaskStatus(dbTask.Status) == codersdk.TaskStatusInitializing {
+		message := "Initializing workspace"
+
+		switch {
+		case ws.LatestBuild.Status == codersdk.WorkspaceStatusPending:
+			message = "Workspace build is pending"
+		case ws.LatestBuild.Status == codersdk.WorkspaceStatusStarting:
+			message = "Starting workspace"
+		case taskAgentLifecycle != nil:
+			switch *taskAgentLifecycle {
+			case codersdk.WorkspaceAgentLifecycleCreated:
+				message = "Agent is connecting"
+			case codersdk.WorkspaceAgentLifecycleStarting:
+				message = "Agent is starting"
+			default:
+				message = "Initializing workspace agent"
+			}
```

Comment:
Should we take into account the coder app as well?

Reply:
> I was thinking about WorkspaceAppHealth, specifically WorkspaceAppHealthInitializing. If disabled we can ignore it.

Ah, I see. So this would be the case where the agent is ready but the app is still initializing, correct? Do you think it makes sense to get into that level of detail? We have different possible AppHealth statuses:
```
WorkspaceAppHealthDisabled     WorkspaceAppHealth = "disabled"
WorkspaceAppHealthInitializing WorkspaceAppHealth = "initializing"
WorkspaceAppHealthHealthy      WorkspaceAppHealth = "healthy"
WorkspaceAppHealthUnhealthy    WorkspaceAppHealth = "unhealthy"
```
I'm guessing only the `initializing` here would make sense, because the others would probably affect the overall task status, correct?

Reply:
> So this would be the case where the agent is ready but the app is still initializing, correct?

Yep! 👍🏻

> Do you think it makes sense to get into that level of detail?

It would better reflect how we determine the tasks status in the `tasks_with_status` view.

> I'm guessing only the `initializing` here would make sense, because the others would probably affect the overall task status, correct?

Correct. 👍🏻

We could consider `unhealthy` as well, but that would open a can of worms I'd prefer to avoid for now (messages for agent health, workspace build failure, etc).

Reply:
Addressed in https://github.com/coder/coder/pull/20692/commits/b1cb7fee56d395c12b508f43858cbcd3ec403e3b